generator/C/include_v1.0/mavlink_helpers.h

```diff
@@ -430,32 +431,33 @@ MAVLINK_HELPER uint8_t mavlink_frame_char_buffer(mavlink_message_t* rxmsg,
  * 2 (MAVLINK_FRAMING_INCOMPLETE, MAVLINK_FRAMING_OK or MAVLINK_FRAMING_BAD_CRC)
  *
  * Messages are parsed into an internal buffer (one for each channel). When a complete
- * message is received it is copies into *returnMsg and the channel's status is
- * copied into *returnStats.
+ * message is received it is copies into *r_message and the channel's status is
+ * copied into *r_mavlink_status.
  *
  * @param chan     ID of the current channel. This allows to parse different channels with this function.
```

Comment:
More language bugs...

Reply:
I've tidied the wording, but I still don't find the explanation _useful_.

What I "think" is that a channel represents a logical connection to an desired endpoint - e.g. a particular GCS or a mavlink camera. Basically anything that needs a specific set of messages streamed to it.  The channel is then just a handle to that logical "thing" for stats and the message signature. 
The text says that it isn't about serial port, but generally I can't see how you would be able to differentiate any incoming information from multiple channels unless they were separated by physical or logical port.

If you think that's about right, please let me know. Shouldn't stop this merging though.

Reply:
channels aren't necessarily *to* endpoints.  Sometimes channels aren't even used for actual communication, just fiddling with messages.  And often there are multiple system IDs that a channel may correspond to.

I usually think about channels as "parallel channels of communication".  Each channel has separate space for incoming and outgoing stuff.  chan just switches between them (it really is just an offset at heart....)

Reply:
Hi @peterbarker ,

> channels aren't necessarily _to_ endpoints. Sometimes channels aren't even used for actual communication, just fiddling with messages.

Can you give me an example?

> And often there are multiple system IDs that a channel may correspond to.

What do you mean. Shouldn't each physical system have just one id? Or are you talking about destinations - ie a channel might talk to multiple endpoints specifically (in addition to broadcasting). 

The [official docs](https://mavlink.io/en/mavgen_c/#streams) throw in the term "stream", but seem to mean the same thing as you indicated.

> The C MAVLink library utilizes a "channel" metaphor to allow for simultaneous processing of multiple, independent MAVLink streams in the same program. It is therefore important to use the correct channel for each operation as all receiving and transmitting functions provided by MAVLink require a channel.
>
> If only one MAVLink stream exists, channel 0 should be used by specifying the MAVLINK_COMM_0 constant.

So what would you see as a logical stream? Can you help me with some examples of things that it makes sense to split up into logical independent handling. And the converse, why not just use one channel?